### DISCUSSION TITLE
testing.make_empty_ppi_radar for multiple elevation scans

### DISCUSSION THREAD

**Discussion by rwall2** [ARM-DOE/pyart#1252] [Q&A] [answered]
I need to create a radar object from a data source not compatible with pyart's IO methods. I've seen this tutorial: https://arm-doe.github.io/pyart/notebooks/basic_ingest_using_test_radar_object.html. The tutorial makes sense for a single elevation scan. But I can't seem to figure out how to create a radar object with multiple elevation scans. Is there another example somewhere for multiple elevation scans?

**Comment by zssherman** [comment]
Hello @rwall2 !

The radar object allows a sweep start and end index dict to be plugged into the radar class a tells pyart where sweep starts and end:
https://github.com/ARM-DOE/pyart/blob/e02ab59b8c792578eca469dad539f340500b1698/pyart/core/radar.py#L62

Examples of this are:
sweep_start_ray_index = {
{'long_name': 'Index of first ray in sweep, 0-based',
 'units': 'count',
 'data': array([   0,  720, 1440, 2160, 2880, 3240, 3600, 3960, 4320, 4680, 5040,
        5400, 5760, 6120, 6480, 6840])}
 
sweep_end_ray_index = {
{'long_name': 'Index of last ray in sweep, 0-based',
 'units': 'count',
 'data': array([ 719, 1439, 2159, 2879, 3239, 3599, 3959, 4319, 4679, 5039, 5399,
        5759, 6119, 6479, 6839, 7199])}

If this information is available along with your fixed angle, you should be able to plug those into the radar class as a dict or make a dict to do so. We are also always looking for new readers etc in Py-ART, and are willing to work with the community to do so.

**Reply by rwall2** [reply]
Thanks. If I have a volume array of 3 dimensions (number of rays, number of bins, number of elevations) (for example (720, 1832, 16) ), how would I reformat the volume array such that pyart can ingest the volume array correctly in addition to using your example code?

**Comment by zssherman** [comment]
Just making sure, so 720 rays per sweep and there are 16 sweeps?

**Reply by rwall2** [reply]
Yes. And 1832 bins (i.e. range gates) per ray.

**Comment by zssherman** [comment, marked as answer]
Gotcha so having uniform rays per sweep makes this slightly easier, so in your case:
radar = make_empty_ppi_radar(1832, 720, 16):

Also in your case:
radar.sweep_start_ray_index['data'] = np.array([0, 720, 1440, 2160, 2880, 3600, 4320, 5040, 5760, 6480, 7200, 7920, 8640, 9360, 10080, 10800])}
radar.sweep_end_ray_index['data'] =  np.array([719, 1439, 2159, 2879, 3599, 4319, 5039, 5759. 6479, 7199, 7919. 8639. 9359, 10079. 10799, 11519])}
this allows pyart to know which ray index the sweeps start and end.

the rest of the variables in the example notebook, you would just have to plug in the arrays for the data attributes.
